src/game/AI/ScriptDevAI/base/BossAI.cpp

```diff
@@ -32,10 +32,22 @@ void BossAI::AddOnAggroText(uint32 text)
     m_onAggroTexts.push_back(text);
 }
 
+void BossAI::Reset()
+{
+    CombatAI::Reset();
+    m_creature->SetSpellList(m_creature->GetCreatureInfo()->SpellList);
+}
+
 void BossAI::JustDied(Unit* killer)
 {
+    CombatAI::JustDied(killer);
     if (!m_onKilledTexts.empty())
         DoBroadcastText(m_onKilledTexts[urand(0, m_onKilledTexts.size() - 1)], m_creature, killer);
+    for (QueuedCast& cast : m_castOnDeath)
```

Comment:
Still not sure if this is a good idea or not. I like the abstraction it provides, but I'm not sure if it makes sense to other people